rasa/core/policies/embedding_policy.py

```diff
@@ -1,694 +1,139 @@
-from typing import Any, List, Optional, Text, Dict, Tuple
-
-import rasa.utils.io
-from rasa.core.domain import Domain
-from rasa.core.featurizers import (
-    TrackerFeaturizer,
-    FullDialogueTrackerFeaturizer,
-    LabelTokenizerSingleStateFeaturizer,
-    MaxHistoryTrackerFeaturizer,
+from typing import Any, Dict, Optional, Text
+
+from rasa.core.constants import DEFAULT_POLICY_PRIORITY, DIALOGUE
+from rasa.core.featurizers import TrackerFeaturizer
+from rasa.core.policies.ted_policy import TEDPolicy
+from rasa.constants import DOCS_URL_POLICIES
+from rasa.utils.tensorflow.constants import (
+    LABEL,
+    HIDDEN_LAYERS_SIZES,
+    TRANSFORMER_SIZE,
+    NUM_TRANSFORMER_LAYERS,
+    NUM_HEADS,
+    MAX_SEQ_LENGTH,
+    BATCH_SIZES,
+    BATCH_STRATEGY,
+    EPOCHS,
+    RANDOM_SEED,
+    RANKING_LENGTH,
+    LOSS_TYPE,
+    SIMILARITY_TYPE,
+    NUM_NEG,
+    EVAL_NUM_EXAMPLES,
+    EVAL_NUM_EPOCHS,
+    NEG_MARGIN_SCALE,
+    REGULARIZATION_CONSTANT,
+    SCALE_LOSS,
+    USE_MAX_NEG_SIM,
+    MAX_NEG_SIM,
+    MAX_POS_SIM,
+    EMBED_DIM,
+    DROPRATE_DIALOGUE,
+    DROPRATE_LABEL,
+    DROPRATE_ATTENTION,
+    KEY_RELATIVE_ATTENTION,
+    VALUE_RELATIVE_ATTENTION,
+    MAX_RELATIVE_POSITION,
 )
-from rasa.core.policies.policy import Policy
-from rasa.core.constants import DEFAULT_POLICY_PRIORITY
-from rasa.core.trackers import DialogueStateTracker
-from rasa.utils import train_utils
-
-import tensorflow as tf
-
-# avoid warning println on contrib import - remove for tf 2
 from rasa.utils.common import raise_warning
+from rasa.utils.tensorflow.models import RasaModel
 
-tf.contrib._warning = None
 logger = logging.getLogger(__name__)
 
 
-class EmbeddingPolicy(Policy):
+class EmbeddingPolicy(TEDPolicy):
     """Transformer Embedding Dialogue Policy (TEDP)
 
-    Transformer version of the REDP used in our paper https://arxiv.org/abs/1811.11707
+    The policy used in our paper https://arxiv.org/abs/1910.00486
     """
 
-    SUPPORTS_ONLINE_TRAINING = True
-
     # default properties (DOC MARKER - don't remove)
     defaults = {
         # nn architecture
-        # a list of hidden layers sizes before user embed layer
+        # a list of hidden layers sizes before dialogue and action embed layers
         # number of hidden layers is equal to the length of this list
-        "hidden_layers_sizes_pre_dial": [],
-        # a list of hidden layers sizes before bot embed layer
-        # number of hidden layers is equal to the length of this list
-        "hidden_layers_sizes_bot": [],
+        HIDDEN_LAYERS_SIZES: {DIALOGUE: [], LABEL: []},
         # number of units in transformer
-        "transformer_size": 128,
+        TRANSFORMER_SIZE: 128,
         # number of transformer layers
-        "num_transformer_layers": 1,
-        # type of positional encoding in transformer
-        "pos_encoding": "timing",  # string 'timing' or 'emb'
+        NUM_TRANSFORMER_LAYERS: 1,
         # max sequence length if pos_encoding='emb'
```

Comment:
where can I set `pos_encoding`?

Reply:
you cannot anymore, also `max_seq_len` will be removed